### PR TITLE
react-accordion-component | Rich Trunzo 

### PR DESCRIPTION
**Pull request by richtrunzo**
https://user-images.githubusercontent.com/69181727/105422397-88cbb280-5bf8-11eb-9411-40dd3a97e3b5.mp4

### CODE REVIEW

File: react-accordion-component/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            document written in a markup language like HTML.
+            CSS is a cornerstone technology of the World Wide
+            Web, alongside HTML and JavaScript`,
+    js: `JavaScript, often abbreviated as JS, is a high - level,
+          interpreted programming language that conforms to
+          the ECMAScript specification JavaScript has curly -
+          bracket syntax, dynamic typing, prototype - based
+          obiect - orientation, and first - class functions.`
+  }
+]
```
Comment (scbowler) [inline react-accordion-component/src/index.jsx]: Here you should have an `array` of objects, 1 object should represent 1 item in the accordion. It should follow something similar to the following layout:

```JAVASCRIPT
const items = [
  {
    title: 'HTML',
    content: 'This is the long content about HTML',
    id: 1
  },
  {
    title: 'JavaScript',
    content: 'Here is some stuff about JS',
    id: 2
  }
];
```

You will of course need to add all the items and the correct content

File: react-accordion-component/src/accordion.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+  cssOn(){
+    this.setState({html: false, css: true, js: false})
+  }
+  htmlOn(){
+    this.setState({html: true, css: false, js: false})
+  }
+  start() {
+    this.setState({ html: false, css: false, js: false })
+  }
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx]: This is way too specific, you should only have one click handler function. You want your accordion to work with any data passed in that is formatted correctly.

A better approach to the accordion is to store only the `id` of the open item in state.

File: react-accordion-component/src/accordion.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {html: false, css: false, js: false};
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx]: Here you only need one property in state, it should just hold the `id` of the open item.

File: react-accordion-component/src/accordion.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+              <p>HyperText Markup Language</p>
+            </div>
+            <div class="lang" onClick={this.cssOn}>
+              <p>Cascading Style Sheets</p>
+            </div>
+            <div class="lang" onClick={this.jsOn}>
+              <p>JavaScript</p>
+            </div>
+            </div>
+  }
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx]: Once again this is too specific, you should be using a `map` method in `render` to loop through your items to create each item.

File: react-accordion-component/src/accordion.jsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    if (this.state.html === true) {
+      return this.htmlRender();
+    } else if (this.state.css === true) {
+      return this.cssRender();
+    } else if (this.state.js === true) {
+      return this.jsRender();
+    } else {
+      return this.startRender();
+    }
+  }
```
Comment (scbowler) [inline react-accordion-component/src/accordion.jsx]: You don't need any conditions here to determine what to render, your render method should look something like:

```JSX
render() {
  return (
    <div className="container">
      {this.props.items.map(item => {
        let contentClass = 'hide';
        
        if(this.state.id === item.id) {
          contentClass = 'show';
        }
        return (
          <div>
            <h4>{item.title}</h1>
            <p className={contentClass}>{item.content}</p>
          </div>
        )
      })}
    </div>
  );
  
  ```
  Notice we can use the id to compare with the id in state to determine which item should be open. This is not a full solution and probably doesn't match your variable and class names but it is to point you in the right direction.

**Review by scbowler — CHANGES_REQUESTED** [review]
When building things with code we want them to be as dynamic as possible not very specific to one data set. I have left some feedback on ways you can make your code more dynamic. Let us know if you have any questions or need any clarification on any of the feedback.

**Comment by richtrunzo** [comment]
Updated per your feedback @scbowler - new video is below! 


https://user-images.githubusercontent.com/69181727/105449583-86d01680-5c2d-11eb-82f5-dd9bb7a479fb.mp4

**Review by BrettAlbright — APPROVED** [review]
👍

**Pull request merged**